### PR TITLE
Encryption support on UWP

### PR DESCRIPTION
**Pull request by sergio-nsk**
I am stuck on PKCS7 signing. Let keep BCrypt and NCrypt w/o `mz_crypt_sign()` and `mz_crypt_sign_verify()` in this PR.
Also I am stuck on building the tests for UWP. It sucks on application manifest files, a manifest file in one project breaks build of another project with another manifest file.

### CODE REVIEW

File: mz_crypt_winrt.c [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+    }
+    if (NT_SUCCESS(status))
+        return size;
+
+    return mz_os_rand(buf, size);
+}
+
+/***************************************************************************/
+
+/* Adapted from RFC4634 &Igor Pavlov's 2010 public domain implementation */
```
Comment (nmoinvaz) [inline mz_crypt_winrt.c:41]: I think there is an & ampersand that is a typo..?

Comment (nmoinvaz) [inline mz_crypt_winrt.c:41]: I think there is an & ampersand that is a typo..?

File: test/test_crypt.cc [hunk 2]
```diff
@@ -138,32 +138,83 @@ TEST(crypt, sha512) {
...
+    uint8_t buf[120];
+
+    key_length = (int32_t)strlen(key);
+    test_length = (int32_t)strlen(test);
+
+    strncpy((char *)buf, test, sizeof(buf));
+
+    mz_crypt_aes_create(&aes);
+    mz_crypt_aes_set_encrypt_key(aes, key, key_length);
+    EXPECT_EQ(mz_crypt_aes_encrypt(aes, buf, test_length), 16);
```
Comment (nmoinvaz) [inline test/test_crypt.cc:184]: I think for 194-bit test, the key length must be 24 bytes?

File: test/test_crypt.cc [hunk 3]
```diff
@@ -138,32 +138,83 @@ TEST(crypt, sha512) {
...
 
     key_length = (int32_t)strlen(key);
     test_length = (int32_t)strlen(test);
 
     strncpy((char *)buf, test, sizeof(buf));
 
     mz_crypt_aes_create(&aes);
-    mz_crypt_aes_set_mode(aes, MZ_AES_ENCRYPTION_MODE_256);
     mz_crypt_aes_set_encrypt_key(aes, key, key_length);
     EXPECT_EQ(mz_crypt_aes_encrypt(aes, buf, test_length), 16);
```
Comment (nmoinvaz) [inline test/test_crypt.cc:212]: I think for 194-bit test, the key length must be 32 bytes?

File: test/test_crypt.cc (hunk 2 above)
Comment (sergio-nsk) [reply test/test_crypt.cc:184]: No, it must not. 16 is a plain data size.

File: test/test_crypt.cc (hunk 3 above)
Comment (sergio-nsk) [reply test/test_crypt.cc:212]: No, it must not. 16 is a plain data size in `buf`.

Comment (nmoinvaz) [reply test/test_crypt.cc:212]: Keylength should be 32? Maybe I commented on the wrong line. Because it will use the key length to determine the AES mode.

Comment (sergio-nsk) [reply test/test_crypt.cc:212]: The key length is 32
```
awesomekeythisisevenmoresolidone
```
Split by 10
```
awesomekey-thisiseven-moresolido-ne
```

Comment (nmoinvaz) [reply test/test_crypt.cc:212]: Ah thanks for pointing that out.

File: CMakeLists.txt [hunk 4]
```diff
@@ -25,6 +29,7 @@ option(MZ_FORCE_FETCH_LIBS "Enables fetching third-party libraries always" OFF)
 option(MZ_PKCRYPT "Enables PKWARE traditional encryption" ON)
 option(MZ_WZAES "Enables WinZIP AES encryption" ON)
 option(MZ_OPENSSL "Enables OpenSSL for encryption" ${UNIX})
+option(MZ_BCRYPT "Enables BCrypt for encryption" ${WIN32UWP})
```
Comment (nmoinvaz) [inline CMakeLists.txt:32]:
```cmake
option(MZ_BCRYPT "Enables BCrypt for encryption" (CMAKE_SYSTEM_NAME STREQUAL "WindowsStore"))
```

Comment (nmoinvaz) [reply CMakeLists.txt:32]: `MZ_BCRYPT` also needs to be documented in the `README.md`.

File: CMakeLists.txt [hunk 5]
```diff
@@ -432,14 +437,19 @@ if(WIN32)
     list(APPEND MINIZIP_DEF -D_CRT_SECURE_NO_DEPRECATE)
     list(APPEND MINIZIP_SRC mz_os_win32.c mz_strm_os_win32.c)
 
-    if(MZ_PKCRYPT OR MZ_WZAES OR MZ_SIGNING)
+    if(MZ_PKCRYPT OR MZ_WZAES OR MZ_SIGNING OR MZ_BCRYPT)
```
Comment (nmoinvaz) [inline CMakeLists.txt:440]: I'm not sure we need `MZ_BCRYPT` check here. Crypt is always conditional of `MZ_PKCRYPT OR MZ_WZAES`

**Review by nmoinvaz — COMMENTED** [review]
Only a few feedback..

File: CMakeLists.txt (hunk 4 above)
Comment (sergio-nsk) [reply CMakeLists.txt:32]: > Does this work?

IDK, need try it.

Comment (sergio-nsk) [reply CMakeLists.txt]: Made it a dependent option, forced ON on WindowsStore and user choice otherwise.

File: CMakeLists.txt (hunk 5 above)
Comment (sergio-nsk) [reply CMakeLists.txt]: Removed and checked.

**Review by nmoinvaz — APPROVED** [review]

**Pull request merged**